### PR TITLE
Iterate through on screen/visible bodies and lights

### PR DESCRIPTION
**Pull request by giann**
Avoid costly loops over large amount of bodies/lights by maintaining a list of visible/on screen bodies and lights in update and iterating through them in drawShadows.

### CODE REVIEW

File: lib/util.lua [hunk 1]
```diff
@@ -1,9 +1,19 @@
 local util = {}
 --TODO: the whole stencil/canvas system should be reviewed since it has been changed in a naive way
 
+local temp = nil
+
 function util.process(canvas, options)
   --TODO: now you cannot draw a canvas to itself  
-  temp = love.graphics.newCanvas()
+  if not temp then
```
Comment (tanema) [inline lib/util.lua]: Maybe this change should be done in PR #28 where discussion is being held on it? Could you take out this commit and either show this code as example in the other PR or open a PR with this change alone?

**Comment by punk0x29a** [comment]
It's great! It actually works like two times faster in my project. Good job!

**Comment by F483** [comment]
Does this break anything? If not is there a reason it has not been merged into master? I would prefer to be running a stable master over a collection of pull requests I had to merge myself.

**Comment by giann** [comment]
No reason apart from me not taking the time to address @tanema comment

File: lib/util.lua (hunk 1 above)
Comment (ScottHerrmann) [reply lib/util.lua]: It looks like PR #28 is clean, if we can get it commited and have him remove this it should be good to commit yes?

**Pull request closed without merging**